python/src/trezorlib/definitions.py

```
@@ -155,12 +232,39 @@ def fetch_path(self, *components: str) -> t.Optional[bytes]:
+            index = name.find(marker)
+            if index != -1 and (index == 0 or name[index - 1] == "/"):
+                return name[:index]
+        return ""
 
     def fetch_path(self, *components: str) -> t.Optional[bytes]:
-        inner_name = "/".join(components)
+        inner_name = self.root + "/".join(components)
         LOG.info("Extracting definition from %s:%s", self.archive.name, inner_name)
         try:
             return self.archive.extractfile(inner_name).read()  # type: ignore [not a known attribute]
         except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)